test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -220,6 +227,7 @@
     private Function<Client, Client> clientWrapper;
 
     public InternalTestCluster(String nodeMode, long clusterSeed, Path baseDir,
+                               int numMasterNodes,
```

Comment:
I think if we could simplify this a bit, like having a simple boolean to enable or disable dedicated masters, would make things a bit easier to understand. I personally find numMasterNodes = 0 counter intuitive, I have to think about what it does for around 10 seconds every time I see it :) also the interaction between numDataNodes and numMasterNodes makes things a bit complicated, but that we could improve later too.